client/packages/system/src/Name/ListView/ImportProperties/ImportReviewDataTable.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { FC, useState } from 'react';
+import {
+  ColumnDescription,
+  DataTable,
+  Grid,
+  NamePropertyNode,
+  NothingHere,
+  Pagination,
+  PropertyNode,
+  SearchBar,
+  TooltipTextCell,
+  useColumns,
+  useTranslation,
+} from '@openmsupply-client/common';
+import { ImportRow } from './PropertiesImportModal';
+
+interface ImportReviewDataTableProps {
+  rows: ImportRow[];
+  properties: NamePropertyNode[] | undefined;
+}
+export const ImportReviewDataTable: FC<ImportReviewDataTableProps> = ({
+  rows,
+  properties,
+}) => {
+  const t = useTranslation();
+  const [pagination, setPagination] = useState<Pagination>({
+    page: 0,
+    first: 20,
+    offset: 0,
+  });
+  // Could filter here for only properties that are used in import
+  const propertyNodes: PropertyNode[] | undefined = properties
+    ?.map(property => {
+      return { ...property.property };
+    })
+    .sort();
+
+  const [searchString, setSearchString] = useState<string>(() => '');
+  const columnDescriptions: ColumnDescription<ImportRow>[] = [
+    {
+      key: 'code',
+      width: 50,
+      sortable: false,
+      label: 'label.code',
+    },
+    {
+      key: 'name',
+      width: 150,
+      sortable: false,
+      label: 'label.name',
+      Cell: TooltipTextCell,
+    },
+  ];
+  propertyNodes?.map(property =>
+    columnDescriptions.push({
+      key: property.key,
+      width: 100,
+      sortable: false,
+      label: undefined,
+      labelProps: { defaultValue: property.name },
+      Cell: TooltipTextCell,
+    })
+  );
+
+  {
+    columnDescriptions.push({
+      key: 'errorMessage',
+      label: 'label.error-message',
+      width: 150,
+      Cell: TooltipTextCell,
+    });
+  }
+
+  const rowsWithProperties = rows.map(row => {
+    return { ...row, ...row.properties };
+  });
+
+  const columns = useColumns<ImportRow>(columnDescriptions, {}, []);
+
+  const filteredEquipment = rowsWithProperties.filter(row => {
+    if (!searchString) {
+      return true;
+    }
+    return (
+      row.name.includes(searchString) ||
+      (row.code && row.code.includes(searchString)) ||
+      (row.errorMessage && row.errorMessage.includes(searchString)) ||
+      row.id === searchString
+    );
+  });
+  const currentFacilitiesPage = filteredEquipment.slice(
+    pagination.offset,
+    pagination.offset + pagination.first
+  );
+
+  // console.log('currentFacilitiesPage', currentFacilitiesPage);
+
+  return (
+    <Grid flexDirection="column" display="flex" gap={0}>
```

Comment:
<img width="1257" alt="Screenshot 2024-06-25 at 10 05 29 AM" src="https://github.com/msupply-foundation/open-msupply/assets/55115239/cf11bb71-45f3-4e4f-9397-48331067e327">

Something's up with styles here, getting overflow issues in both directions (can't scroll)

Reply:
This one is tricky and I wonder if needs a larger refactor.

It looks like the table is basing width off the parent page rather than the modal, so doesn't think it needs a scroll.

I've added a width override, but I wonder if we need a better way to handle table width when on modals